tests/operators/sensors.py

```diff
@@ -12,11 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+import logging
 import os
+import time
 import unittest
 
-from airflow.operators.sensors import HttpSensor
-from airflow.exceptions import AirflowException
+from datetime import datetime, timedelta
+
+from airflow import DAG, configuration
+from airflow.operators.sensors import HttpSensor, BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.exceptions import (AirflowException,
+                                AirflowSensorTimeout,
+                                AirflowSkipException)
+configuration.test_mode()
+
+DEFAULT_DATE = datetime(2015, 1, 1)
+TEST_DAG_ID = 'unit_test_dag'
+
+
+class TimeoutTestSensor(BaseSensorOperator):
+    '''
+    Sensor that always returns the return_value provided
+
+    :param return_value: Set to true to mark the task as SKIPPED on failure
+    :type return_value: any
+    '''
+    ui_color = '#e6f1f2'
+
+    @apply_defaults
+    def __init__(
+            self,
+            return_value=False,
+            *args,
+            **kwargs):
+        self.return_value = return_value
+        super(TimeoutTestSensor, self).__init__(*args, **kwargs)
+
+    def poke(self, context):
+        return self.return_value
+
+    def execute(self, context):
+        started_at = datetime.now()
+        time_jump = self.params.get('time_jump')
+        while not self.poke(context):
+            time.sleep(self.poke_interval)
+            if time_jump:
+                started_at -= time_jump
+            if (datetime.now() - started_at).total_seconds() > self.timeout:
+                if self.soft_fail:
+                    raise AirflowSkipException('Snap. Time is OUT.')
+                else:
+                    raise AirflowSensorTimeout('Snap. Time is OUT.')
+        logging.info("Success criteria met. Exiting.")
+
+
+class SensorTimeoutTest(unittest.TestCase):
+    def setUp(self):
+        configuration.test_mode()
+        args = {
+            'owner': 'airflow',
+            'start_date': DEFAULT_DATE
+        }
+        dag = DAG(TEST_DAG_ID, default_args=args)
+        self.dag = dag
+
+    def test_timeout(self):
+        t = TimeoutTestSensor(
+            task_id='test_timeout',
+            execution_timeout=timedelta(days=2),
+            return_value=False,
+            poke_interval=5,
```

Comment:
can we decrease this to it's minimum value to reduce the time impact on the test suite? Also consider adding a TODO to mock out the time objects so that we don't have to actually wait <interval> seconds (sleeping in tests is generally an antipattern).